app/controllers/split_checkout_controller.rb

```diff
@@ -68,6 +69,11 @@ def advance_order_state
     OrderWorkflow.new(@order).advance_checkout(raw_params.slice(:shipping_method_id))
   end
 
+  def validate_current_step!
+    step = params[:step].tap{ |step| ["details", "payment", "summary"].include? step }
```

Comment:
I don't understand the `tap` call here. It doesn't change anything. Does it need to `raise`?

Reply:
Ah right I was trying to be concise but it's actually not filtering it :see_no_evil: 

This was a last-minute addition.

How about using the intersection operator...
```suggestion
    step = ([params[:step]] & ["details", "payment", "summary"]).first
```

Reply:
I like that syntax.

Maybe it would also be clearer to raise an error explicitly instead of trying to call `validate_!`. But it will fail and not call any unintended code anyway.

Reply:
Shorter but more cryptic:

```suggestion
    step, = [params[:step]] & ["details", "payment", "summary"]
```
Probably not worth the change.

Reply:
That is incredibly cryptic! :exploding_head:

Reply:
I don't know if i like it ;) 🧠